eva/catalog/services/udf_profile_service.py

```diff
@@ -0,0 +1,83 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from sqlalchemy.orm.exc import NoResultFound
+
+from eva.catalog.models.udf_profile import UdfProfileMetadata
+from eva.catalog.services.base_service import BaseService
+from eva.utils.logging_manager import logger
+from eva.utils.metrics import Metrics
+
+
+class UdfProfileService(BaseService):
+    def __init__(self):
+        super().__init__(UdfProfileMetadata)
+
+    def create_udf_profile(self, udf_id: int, metrics: Metrics) -> UdfProfileMetadata:
+        """Creates a new udf profile entry
+
+        Arguments:
+            udf_id (str): udf_id corresponding to the metrics
+            metrics: Metrics corresponding to the udf
+
+        Returns:
+            UdfMetadata: Returns the new entry created
+        """
+        metadata = self.model(udf_id,
+                              metrics.batch_size,
+                              metrics.time_taken,
+                              metrics.accuracy)
+        metadata = metadata.save()
+        self.print_all_profile("Post Inserting {}".format(udf_id))
```

Comment:
Unsure how to print catalog table as of now. Using print statements to ensure correctness.